packages/react/src/index.ts

```diff
@@ -64,13 +66,39 @@ function setCurrentUpdater(updater?: Effect) {
 	finishUpdate = updater && updater._start();
 }
 
-function createUpdater(update: () => void) {
+/**
+ * A redux-like store whose store value is a positive 32bit integer (a 'version') to be used with useSyncExternalStore API.
+ * React (current owner) subscribes to this store and gets a snapshot of the current 'version'.
+ * Whenever the 'version' changes, we tell React it's time to update the component (call 'onStoreChange').
+ *
+ * How we achieve this is by creating a binding with an 'effect', when the `effect._callback' is called,
+ * we update our store version and tell React to re-render the component ([1] We don't really care when/how React does it).
+ *
+ * [1]
+ * @see https://reactjs.org/docs/hooks-reference.html#usesyncexternalstore
+ * @see https://github.com/reactjs/rfcs/blob/main/text/0214-use-sync-external-store.md
+ */
+function createEffectStore() {
 	let updater!: Effect;
+	let version = 0;
+
 	effect(function (this: Effect) {
 		updater = this;
 	});
-	updater._callback = update;
-	return updater;
+
+	return {
+		updater,
+		subscribe(onStoreChange: () => void) {
+			updater._callback = function () {
+				version = (version + 1) & ~(1 << 31);
+				onStoreChange();
+			};
+			return function _unsubscribe() {};
```

Comment:
We should probably clean up the `effect()` here. Every `effect()` call returns an unsubscribe functions so we can return that one here.

```js
const unsubscribe = effect(() => ...);
```

Reply:
I did so at first, but cleaning up the effect makes the test `should consistently rerender in strict mode` fail.
The previous implementation didn't unsubscribe either so I thought maybe it was intentional 🤔

Reply:
Is it possible the useMemo() is running twice in strict mode? I'm not up-to-date on the funky semantics of React's modes, but afaik that would cause unsubscription and maybe wouldn't re-subscribe properly?

Reply:
@developit yes, please check this commit: https://github.com/preactjs/signals/pull/219/commits/2e9ea59ce63ec10be84703b31a88e7491cc4cb7e

I'm not sure if this'd be the best approach though. Another way could be to rely on the React current owner internal (which is set only on dev) but I tried to make it rely as less as possible on the internals.

I left comments in the code. I tried my best to explain it 😅 
In prod, it'll always unsubscribe, on dev it lazily unsubscribes itself. This can be tested by setting `__DEV__ = false` in which case the StrictMode is the only test that'll fail.

I'm also not quite sure how you'd test using the React prod entry point, apparently if tests are executed with `NODE_ENV=production` this `act` thing errors with kinda "_isn't supported with production build_". Do these Netlify demos use the production one?